apps/snitch_core/test/domain/order/transitions_test.exs

```diff
@@ -199,4 +219,125 @@ defmodule Snitch.Domain.Order.TransitionsTest do
 
     assert result.valid?
   end
+
+  describe "process payment for order with one package" do
+    setup :zones
+    setup :shipping_methods
+    setup :embedded_shipping_methods
+    setup :payment_methods
+
+    setup %{embedded_shipping_methods: methods} do
+      order = insert(:order, user: build(:user), total: Money.new(3, :USD))
+
+      [order: order, packages: [insert(:package, shipping_methods: methods, order: order)]]
+    end
+
+    @tag shipping_method_count: 1
+    test "with valid card details", %{order: order, packages: [package], shipping_methods: [sm]} do
+      expect(Snitch.Tools.DefaultsMock, :fetch, fn :currency -> {:ok, :USD} end)
```

Comment:
1. Use the `expect/4` clause, save a few lines.
2. Why are 2 transitions being tested here?